src/server/req_quejob.c

```diff
@@ -1719,7 +1716,7 @@ req_commit_now(struct batch_request *preq,  job *pj)
 	pj->ji_qs.ji_un.ji_momt.ji_exitstat = 0;
 	if ((pj->ji_qs.ji_svrflags & (JOB_SVFLG_CHKPT|JOB_SVFLG_ChkptMig)) == 0) {
 		pj->ji_qs.ji_stime = time_now; 	/* start of walltime */
-		pj->ji_wattr[(int)JOB_ATR_stime].at_flags |= ATR_VFLAG_MODIFY;
+		set_jattr_l_slim(pj, JOB_ATR_stime, time_now, SET);
```

Comment:
I think we need to remove ji_stime from quick save area (I'm surprised i hadn't removed it myself). Otherwise, we'll end up only setting ji_stime directly somewhere outside of the setters and we will miss the time when stime was set on the job. We should remove it and replace all calls to it with calls to getters/setters for JOB_ATR_stime

Reply:
This PR at its current state itself is quite huge. And as we discussed sometime back we both agree it would have been easier to have this done in small chunks. Maybe we can tackle it later with memory reduction work.